kikit/fab/jlcpcb.py

```diff
@@ -71,7 +72,12 @@ def exportJlcpcb(board, outputdir, assembly, schematic, ignore, field,
     gerberdir = os.path.join(outputdir, "gerber")
     shutil.rmtree(gerberdir, ignore_errors=True)
     gerberImpl(board, gerberdir)
-    archiveName = nametemplate.format("gerbers")
+
+    textVars = kikitTextVars(loadedBoard)
+    try:
+        archiveName = nametemplate.format("gerbers", **textVars)
+    except KeyError as e:
+        raise RuntimeError(f"Unknown variable {e} in nametemplate:\n{nametemplate}")
```

Comment:
This block appears identically in multiple places. Please, factor the code out into a separate function.

Reply:
I will think about something, should we pass the board to the function and call `kikitTextVars` inside the newly created function or do you believe is sufficient to pass only the `kikitTextVars` dict?

Reply:
I suggest:
```.py
def expandNametemplate(template: str, filetype: str, board: pcbnew.BOARD) -> str
```